include/crow/common.h

```diff
@@ -275,6 +275,9 @@ namespace crow
         return string_params[index];
     }
     /// @endcond
+
+    using routing_search_result = std::tuple<uint16_t, std::vector<uint16_t>, routing_params>;
+    using routing_handle_result = std::tuple<uint16_t, std::vector<uint16_t>, routing_params, HTTPMethod>;
```

Comment:
Maybe create structs? It will be more readable to `res->routing_params` than `std::get<2>(var)`.

Reply:
I'd also go for structs.
Maybe a single struct is even applicable as the only difference is `HTTPMethod`, it could be stored in a `unique_ptr`?